packages/api/src/lib/granule-delete.ts

```diff
@@ -63,9 +74,10 @@ const deleteGranuleAndFiles = async ({
     );
 
     await knex.transaction(async (trx) => {
-      // TODO: relying on the cumulus_id from the lookup is icky, but we need to
-      // truly identify the unique record.
-      await granulePgModel.delete(trx, { cumulus_id: pgGranule.cumulus_id });
+      await granulePgModel.delete(trx, {
+        granule_id: pgGranule.granule_id,
+        collection_cumulus_id: pgGranule.collection_cumulus_id,
+      });
```

Comment:
Dumb question: why is this preferable to deleting via the `cumulus_id`? They both should uniquely identify a Granule and the `cumulus_id` is the primary key so all other things being equal (which they are emphatically not 😄 ), that seems like a reasonable thing to do. 

I would think that a compelling reason to change would be if we could avoid looking up the PG Granule ahead of time, which we're not able to do here. I think I'm missing something.

Related and probably invalidated by your response to the above, is there any value in having this work with a `pgGranule.cumulus_id` _or_ `granule_id` + `collection_cumulus_id`?

Reply:
Ultimately, the whole goal of this PR is to enforce safety when doing granule deletes, since they have cascading effects on the files table and the granules/executions join table. So when you do a `granulePgModel.delete` from TS now, you have to specify either an object with `cumulus_id` or an object with `granule_id` + `collection_cumulus_id` to ensure that you are **uniquely identifying a granule record to delete and not deleting multiple granule records**.

> would think that a compelling reason to change would be if we could avoid looking up the PG Granule ahead of time, which we're not able to do here.

At the moment, you're right, there's no compelling reason not to use `cumulus_id` vs the other fields as they both uniquely identify a granule. Doing it this way may allow us to avoid looking up the granule before deletion in the future, but it doesn't provide any value now per so. I can revert to using `cumulus_id` for now if you prefer. Mostly I just wanted to prove that the TS will fail to compile if you don't use a valid set of properties to uniquely identify a granule

> is there any value in having this work with a pgGranule.cumulus_id or granule_id + collection_cumulus_id?

That is exactly how it does work now

Reply:
Ah, ok that makes sense that we're setting ourselves up to eliminate a granule lookup in the future. I buy that.

My second point there about cumulus_id vs. granule_id + collection_cumulus_id was from the perspective of this code in granule-delete. Totally makes sense that the model accepts either. I don't think there's a good reason to change anything here based on the above reasoning.